ops/opset13/cos.go

```diff
@@ -0,0 +1,76 @@
+package opset13
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/advancedclimatesystems/gonnx/onnx"
+	"github.com/advancedclimatesystems/gonnx/ops"
+	"gorgonia.org/tensor"
+)
+
+// Cos represents the ONNX cos operator.
+type Cos struct{}
+
+// newCos creates a new cos operator.
+func newCos() ops.Operator {
+	return &Cos{}
+}
+
+// Init initializes the cos operator.
+func (c *Cos) Init(attributes []*onnx.AttributeProto) error {
+	return nil
+}
+
+type CosDType interface {
+	float32 | float64
+}
+
+// Apply applies the cos operator.
+func (c *Cos) Apply(inputs []tensor.Tensor) ([]tensor.Tensor, error) {
+	var out tensor.Tensor
+	var err error
+	if inputs[0].Dtype() == tensor.Float32 {
+		out, err = inputs[0].Apply(cos[float32])
```

Comment:
This panics if an empty inputs slice is provided?  Or do we validate this somewhere else.

Reply:
We call `op.ValidateInputs` before applying the operator. So in general it could panic but only if the operator is implemented incorrectly and the wrong number of inputs is defined.